src/gt4py/analysis/passes.py

```diff
@@ -67,6 +67,34 @@ def __init__(self, data_type, message=None, *, loc=None):
         super().__init__(message, loc=loc)
 
 
+def _make_parallel_interval_info(parallel_interval):
+    """Create a list of IntervalInfos from a list of AxisIntervals."""
+    if parallel_interval is None:
+        return None
+    else:
+        return [
+            IntervalInfo(
+                start=(interval.start.level, interval.start.offset),
+                end=(interval.end.level, interval.end.offset),
+            )
+            for interval in parallel_interval
+        ]
+
+
+def _make_parallel_interval_from_info(parallel_interval_info):
+    """Create a list of AxisInterval from list of IntervalInfos."""
+    if parallel_interval_info is None:
+        return None
+    else:
+        return [
+            gt_ir.AxisInterval(
+                start=gt_ir.AxisBound(level=interval_info.start[0], offset=interval_info.start[1]),
+                end=gt_ir.AxisBound(level=interval_info.end[0], offset=interval_info.end[1]),
+            )
+            for interval_info in parallel_interval_info
+        ]
+
```

Comment:
Thanks for pointing this out. I can probably it a bit cleaner in a second pass through.

The complexity comes about because the implementation is using `ListOf[IntervalInfo]` in the `IJBlockInfo`:
```python
parallel_interval = attribute(of=ListOf[IntervalInfo], optional=True)
```
These lists are tougher to create with a single statement.